plugins/quickhighlight/pluma-quickhighlight-plugin.c

```diff
@@ -0,0 +1,482 @@
+{
+    g_return_if_fail (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+
+    if (plugin->priv->queued_highlight != 0)
+    {
+        return;
+    }
+
+    plugin->priv->queued_highlight =
+        gdk_threads_add_idle_full (G_PRIORITY_LOW,
+                                   pluma_quick_highlight_plugin_highlight_worker,
+                                   g_object_ref (plugin),
+                                   g_object_unref);
+}
+
+static void
+pluma_quick_highlight_plugin_notify_weak_buffer_cb (gpointer data,
+                                                    GObject *where_the_object_was)
+{
+    PlumaQuickHighlightPlugin *plugin = PLUMA_QUICK_HIGHLIGHT_PLUGIN (data);
+
+    g_assert (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+
+    plugin->priv->style_scheme_handler_id = 0;
+    plugin->priv->buffer = NULL;
+}
+
+static void
+pluma_quick_highlight_plugin_unref_weak_buffer (PlumaQuickHighlightPlugin *plugin)
+{
+    g_return_if_fail (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+
+    if (plugin->priv->buffer == NULL)
+    {
+        return;
+    }
+
+    if (plugin->priv->delete_range_handler_id > 0)
+    {
+        g_signal_handler_disconnect (plugin->priv->buffer,
+                                     plugin->priv->delete_range_handler_id);
+        plugin->priv->delete_range_handler_id = 0;
+    }
+
+    if (plugin->priv->mark_set_handler_id > 0)
+    {
+        g_signal_handler_disconnect (plugin->priv->buffer,
+                                     plugin->priv->mark_set_handler_id);
+        plugin->priv->mark_set_handler_id = 0;
+    }
+
+    if (plugin->priv->style_scheme_handler_id > 0)
+    {
+        g_signal_handler_disconnect (plugin->priv->buffer,
+                                     plugin->priv->style_scheme_handler_id);
+        plugin->priv->style_scheme_handler_id = 0;
+    }
+
+    g_object_weak_unref (G_OBJECT (plugin->priv->buffer),
+                         pluma_quick_highlight_plugin_notify_weak_buffer_cb,
+                         plugin);
+
+    plugin->priv->buffer = NULL;
+}
+
+static void
+pluma_quick_highlight_plugin_set_buffer (PlumaQuickHighlightPlugin *plugin,
+                                         PlumaDocument             *buffer)
+{
+    g_return_if_fail (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+    g_return_if_fail (PLUMA_IS_DOCUMENT (buffer));
+
+    if (plugin->priv->buffer == buffer)
+    {
+        return;
+    }
+
+    pluma_debug (DEBUG_PLUGINS);
+
+    pluma_quick_highlight_plugin_unref_weak_buffer (plugin);
+
+    plugin->priv->buffer = buffer;
+
+    if (plugin->priv->buffer != NULL)
+    {
+        g_object_weak_ref (G_OBJECT (plugin->priv->buffer),
+                           pluma_quick_highlight_plugin_notify_weak_buffer_cb,
+                           plugin);
+
+        plugin->priv->style_scheme_handler_id =
+            g_signal_connect (plugin->priv->buffer,
+                              "notify::style-scheme",
```

Comment:
we usually put the signal name on the same line as the signal origin in g_signal_connect.